api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -138,6 +138,21 @@ default String name() {
    */
   List<HistoryEntry> history();
 
+  /**
+   * Get the snapshot references of this table.
+   *
+   * @return a map with ref name as key, {@link SnapshotRef} as value
+   */
+  Map<String, SnapshotRef> refs();
+
+  /**
+   * Get the snapshot reference with the given name
+   *
+   * @param refName snapshot reference name
+   * @return the SnapShot ref with the given name if it exists, null otherwise.
+   */
+  SnapshotRef ref(String refName);
```

Comment:
nit: prefer simple variable name, `name` instead of `refName` since `ref` is implied from the method name.